lib/logstash/inputs/jdbc.rb

```diff
@@ -232,4 +249,23 @@ def update_state_file
     end
   end
 
+  private
+  # make sure the encoding is uniform over fields
+  def convert(key, value)
+    return value unless should_convert(key, value)
+    converter = find_converter_for value
+    converter.convert(value)
+  end
+
+  def should_convert(key, value)
+    value.is_a?(String) && (@columns_to_encode.empty? || @columns_to_encode.include?(key))
+  end
+
+  def find_converter_for(value)
+    unless @converters.keys.include?(value.encoding)
```

Comment:
Yes, this is the strategy. I think should reliable, but wil do some research to probe it here. Makes sense?